integrators/ordinary_differential_equations.hpp

```diff
@@ -37,9 +42,13 @@ struct SpecialSecondOrderDifferentialEquation {
     std::vector<Displacement> position_error;
     std::vector<Velocity> velocity_error;
   };
+  // A functor that computes f(q, t) and stores it in |*accelerations|.
+  // |*accelerations| must have the same size as |positions|.
```

Comment:
No, that statement about the size is a precondition, not a postcondition, I'll reword to clarify.
There is no precondition on the values stored in `*accelerations` (in the current implementations I think they're just what's left from the previous call).